tests/shell/tests.py

```diff
@@ -94,3 +129,26 @@ def test_shell_with_bpython_not_installed(self, select):
     # returns EOF and so select always shows that sys.stdin is ready to read.
     # This causes problems because of the call to select.select() toward the
     # end of shell's handle() method.
+
+    def test_get_namespace(self):
+        from .models import Marker, Phone
+
+        cmd = shell.Command()
+        imports = cmd.get_namespace().values()
+
+        self.assertIn(Marker, imports)
+        self.assertIn(Phone, imports)
+        self.assertIn(User, imports)
```

Comment:
Better for tests to strictly check everything that is returned, rather than a subset, so compare against the full expected namespace dictionary:

```suggestion
        namespace = cmd.get_namespace()
        
        self.assertEqual(
            namespace,
            {"Marker": Marker, ...}
        )
```

.. and in `test_overridden_get_namespace`

Reply:
I'm having a little problem with this because of some models imported into the namespace, and I'm not sure if they should be. For example, when trying the imports in a separate app, `django.contrib.sites.models.Site` is imported, but in the test case, it isn't. Is this related to the settings of the specific app? What am I missing or saying wrong?